django_mongodb/compiler.py

```diff
@@ -17,15 +19,187 @@ class SQLCompiler(compiler.SQLCompiler):
     """Base class for all Mongo compilers."""
 
     query_class = MongoQuery
+    SEPARATOR = "10__MESSI__3"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._group_pipeline = None
+
+    def _get_group_alias_column(self, col, annotation_group_idx):
+        """Generate alias and replacement for group columns."""
+        replacement = None
+        if not isinstance(col, Col):
+            alias = f"__annotation_group{next(annotation_group_idx)}"
+            col_expr = self._get_column_from_expression(col, alias)
+            replacement = col_expr
+            col = col_expr
+        if self.collection_name == col.alias:
+            return col.target.column, replacement
+        return f"{col.alias}{self.SEPARATOR}{col.target.column}", replacement
+
+    def _get_column_from_expression(self, expr, alias):
+        """Get column target from expression."""
+        column_target = expr.output_field.__class__()
+        column_target.db_column = alias
+        column_target.set_attributes_from_name(alias)
+        return Col(self.collection_name, column_target)
+
+    def _prepare_expressions_for_pipeline(self, expression, target, count):
+        """Prepare expressions for the aggregation pipeline."""
+        replacements = {}
+        group = {}
+        for sub_expr in self._get_aggregate_expressions(expression):
+            alias = f"__aggregation{next(count)}" if sub_expr != expression else target
+            column_target = sub_expr.output_field.__class__()
+            column_target.db_column = alias
+            column_target.set_attributes_from_name(alias)
+            inner_column = Col(self.collection_name, column_target)
+            if sub_expr.distinct:
+                rhs = sub_expr.as_mql(self, self.connection, resolve_inner_expression=True)
+                group[alias] = {"$addToSet": rhs}
+                replacing_expr = sub_expr.copy()
+                replacing_expr.set_source_expressions([inner_column])
+            else:
+                group[alias] = sub_expr.as_mql(self, self.connection)
+                replacing_expr = inner_column
+            # Count must return 0 rather than null.
+            if isinstance(sub_expr, Count):
+                replacing_expr = Coalesce(replacing_expr, 0)
+            # Variance = StdDev^2
+            if isinstance(sub_expr, Variance):
+                replacing_expr = Power(replacing_expr, 2)
+            replacements[sub_expr] = replacing_expr
+        return replacements, group
+
+    def _prepare_annotations_for_group_pipeline(self):
+        """Prepare annotations for the MongoDB aggregation pipeline."""
+        replacements = {}
+        group = {}
+        count = itertools.count(start=1)
```

Comment:
something like `alias_count` or `alias_idx`? Elsewhere the variable is named like `annotation_group_idx` which seems much more descriptive!